src/deny_field.rs

```diff
@@ -0,0 +1,8 @@
+use serde::Deserializer;
+
+pub(crate) fn deny_field<'de, D>(_deserializer: D) -> Result<(), D::Error>
```

Comment:
Could you add a little comment explaining that we can't use `#[serde(deny_unknown_fields)]` because some structs are flattened?